### PR TITLE
Built the reply feature 

### PR DESCRIPTION
**Pull request by izudada**
Built the reply feature  with a fix to the csrf bug

### CODE REVIEW

File: caim_base/templates/animal/view.html [hunk 1]
```diff
@@ -117,9 +117,9 @@ <h4><a href="{{animal.awg.get_absolute_url}}">{{ animal.awg.name }}</a></h4>
     </div>
   </div>
 
-
-
-
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.6.0/jquery.min.js"
```
Comment (electric-al) [inline caim_base/templates/animal/view.html]: JQuery is already included. Do we need this again? Its currently in the footer of the page, so it should work if you use dom ready correctly in the above code.

File: caim_base/static/js/common.js [hunk 2]
```diff
@@ -46,6 +46,47 @@ $('[data-timestamp]').each(function (idx, el) {
     el.text(fmt.format(dte));
 });
 
+$('.reply').click(function(button){
```
Comment (electric-al) [inline caim_base/static/js/common.js]: I think this should really be done in the domready callback? https://api.jquery.com/ready/

File: caim_base/static/js/common.js [hunk 3]
```diff
@@ -46,6 +46,47 @@ $('[data-timestamp]').each(function (idx, el) {
...
+    let buttonID = button.target.id;
+
+        // Reply form element
+    var replyForm = `
+    <form method="post" id="reply-form${buttonID}" class="reply-form" onsubmit="submitForm(event)">
+        <input type="text" name="comment_id" value="${buttonID}" hidden>
```
Comment (electric-al) [inline caim_base/static/js/common.js]: Carried over from previous PR, this should be `<input type="hidden">` not type="text" hidden

**Review by electric-al — CHANGES_REQUESTED** [review]
Almost there! 

Just wondering why jquery is injected again?

File: caim_base/templates/animal/view.html (hunk 1 above)
Comment (izudada) [reply caim_base/templates/animal/view.html]: I noticed a console error. JQuery should always be the topmost tag, else it causes some issues often.

File: caim_base/static/js/common.js (hunk 3 above)
Comment (izudada) [reply caim_base/static/js/common.js]: Great point, will fix this.

File: caim_base/templates/animal/view.html (hunk 1 above)
Comment (electric-al) [reply caim_base/templates/animal/view.html]: If you view the code, you see:

Screenshot 2022-10-31 at 15 41 07

jquery is included twice

File: caim_base/static/js/common.js (hunk 2 above)
Comment (izudada) [reply caim_base/static/js/common.js]: I don't follow. Do you mean I should add the event inside a JQuery-ready function?

File: caim_base/templates/animal/view.html (hunk 1 above)
Comment (izudada) [reply caim_base/templates/animal/view.html]: I see your point. One on ethe wrapper and then the one I added to the view template. WHat do yo suggest?

File: caim_base/views/comments.py [hunk 4]
```diff
@@ -87,3 +91,51 @@ def delete(request, comment_id):
...
             },
         )
+
+
+class CreateSubComment(LoginRequiredMixin, CreateView):
+    model = AnimalSubComment
+    fields = ["body"]
+
+    def form_valid(self, form):
+        if self.request.is_ajax():
```
Comment (electric-al) [inline caim_base/views/comments.py]: https://stackoverflow.com/questions/63629935/django-3-1-and-is-ajax is_ajax is depreciated > django 3.1

File: caim_base/templates/comments/edit_reply.html [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        <div class="container py-5">
+            <h3>Edit Reply</h3>
+
+            <br />
+
+            <form method="post" action="">
+            {% csrf_token %}
+
+                <div class="comment" id="comment-{{comment.id}}">
+                    {% avatar comment.user 40 40 class="comment-avatar" %}
```
Comment (electric-al) [inline caim_base/templates/comments/edit_reply.html]: Should this not be animalsubcomment ?

File: caim_base/templates/comments/delete_reply.html [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+<div class="container">
+    <div class="row justify-content-md-center">
+      <div class="col-md-8">
+        <div class="container py-5">
+            <h3>Are you sure you want to delete this reply?</h3>
+
+            <br />
+
+            <div class="comment" id="comment-{{comment.id}}">
+                {% avatar comment.user 40 40 class="comment-avatar" %}
```
Comment (electric-al) [inline caim_base/templates/comments/delete_reply.html]: Shoudl this be animalsubcomment not comment?

**Review by electric-al — CHANGES_REQUESTED** [review]
One last change:

Screenshot 2022-10-31 at 16 38 26

File: caim_base/templates/comments/edit_reply.html (hunk 5 above)
Comment (izudada) [reply caim_base/templates/comments/edit_reply.html]: Fixed

File: caim_base/templates/comments/delete_reply.html (hunk 6 above)
Comment (izudada) [reply caim_base/templates/comments/delete_reply.html]: fixed

**Pull request merged**